src/envoy/http/alpn/alpn_filter.h

```diff
@@ -22,19 +22,29 @@ namespace Envoy {
 namespace Http {
 namespace Alpn {
 
+using AlpnOverride =
+    absl::flat_hash_map<Http::Protocol, std::vector<std::string>>;
+
 class AlpnFilterConfig {
  public:
-  AlpnFilterConfig() = default;
-  explicit AlpnFilterConfig(
+  AlpnFilterConfig(
       const istio::envoy::config::filter::http::alpn::v2alpha1::FilterConfig
-          &proto_config);
+          &proto_config,
+      Upstream::ClusterManager &cluster_manager);
+
+  Upstream::ClusterManager &clusterManager() { return cluster_manager_; }
 
-  const std::vector<std::string> &getAlpnOverride() const {
-    return alpn_override_;
+  const std::vector<std::string> &alpnOverride(const Http::Protocol &protocol) {
+    return alpn_override_[protocol];
```

Comment:
This is an obvious read only api
Please don't change `alpn_override_`. 
Please 
1. Mark alpnOverride() const
2. return `const std::vector<std::string> * const` since we cannot return reference (alternatively return a default empty string vector else where)